crates/tx5/src/endpoint.rs

```diff
@@ -383,6 +403,132 @@ async fn new_conn_task(
         Ok(r) => r,
     };
 
+    let state_uniq = conn_state.meta().state_uniq.clone();
+    let conn_uniq = conn_state.meta().conn_uniq.clone();
+    let rem_id = conn_state.meta().cli_url.id().unwrap();
+
+    struct Unregister(
+        Option<Box<dyn opentelemetry_api::metrics::CallbackRegistration>>,
+    );
+    impl Drop for Unregister {
+        fn drop(&mut self) {
+            if let Some(mut unregister) = self.0.take() {
+                let _ = unregister.unregister();
+            }
+        }
+    }
+
+    let slot: Arc<std::sync::Mutex<Option<HashMap<String, BackendMetrics>>>> =
```

Comment:
What is the lifetime of this? It looks like it's dropped after `on_new_conn` finishes and with the use of `Weak` neither the task nor the callback will keep it alive

Reply:
Oh I get it, there's a loop further down that I hadn't seen so this function doesn't exit until the connection ends. Thanks for the explanation